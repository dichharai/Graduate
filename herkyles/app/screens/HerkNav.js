import React, {Component} from "react"; 

import Home from "./Home";
import Login from "./Login";
import GymStats from "./GymStats"; 
import Welcome from "./Welcome";  
import AuthLoading from "./AuthLoading";
import DetailedGymInfo from "./DetailedGymInfo"; 
import Signup from './Signup';
import CodeScanner from './CodeScanner';
import RecOptions from './RecOptions';
import RecAreas from './RecAreas';
import AreaEquip from './AreaEquip';
import EquipEntry from './EquipEntry';

import {StackNavigator, SwitchNavigator} from "react-navigation"; 

const AppStack = StackNavigator({ 
    Home: {screen: Home}, 
    Login: {screen: Login}, 
    GymStats: {screen: GymStats},
    DetailedGymInfo: {screen: DetailedGymInfo},
    CodeScanner: {screen: CodeScanner},
    Signup: {screen: Signup},
}); 

const AuthStack = StackNavigator({
    Welcome: {screen: Welcome},
    RecOptions: {screen: RecOptions},
    RecAreas: {screen: RecAreas},
    AreaEquip: {screen: AreaEquip},
    EquipEntry: {screen: EquipEntry}
}); 
// wrapping AppStack & AuthStack in SwitchNavigator so that when a user logs in they aren't shown previously logged in fill-in-page in navigation. 

export default SwitchNavigator(
    {
        AuthLoading: AuthLoading, 
        App: AppStack, 
        Auth: AuthStack, 
    }, 
    {
        initialRouteName: "AuthLoading", 
    }
); 
/*
const RootStack = StackNavigator(
    {
        Home: {screen: Home}, 
        Login: {screen: Login}, 
        GymStats: {screen: GymStats},
        Welcome: {screen: Welcome},  
    },
);
class HerkNav extends Component{
    render(){
        return <RootStack/>
    }
}
export default HerkNav; 
*/
