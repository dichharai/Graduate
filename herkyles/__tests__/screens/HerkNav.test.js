import React from "react";
import HerkNav from "../../app/screens/HerkNav";

import renderer from "react-test-renderer";

describe ("HerkNav", () => {
  it("renders without crashing", () => {
    const rendered = renderer.create(<HerkNav />).toJSON();
    expect(rendered).toBeTruthy();
    expect(rendered).toMatchSnapshot();
  });
});