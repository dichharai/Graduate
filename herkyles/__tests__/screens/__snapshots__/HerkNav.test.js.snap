// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HerkNav renders without crashing 1`] = `
<View
  style={undefined}
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
